core/src/dunit/scala/io/snappydata/cluster/CassandraSnappyDUnitTest.scala

```diff
@@ -269,51 +274,125 @@ class CassandraSnappyDUnitTest(val s: String)
         " repos 'http://clojars.org/repo/'")
     stmt1.execute("list packages")
     rs = stmt1.getResultSet
-    count = 0
-    if (rs ne null) {
-      while (rs.next()) {
-        count += 1
-      }
-      rs.close()
-    }
-    assert(count == 3)
+    assert(getCount(rs) == 3)
+
     stmt1.execute("undeploy mssql")
     stmt1.execute("undeploy cassandrajar")
     stmt1.execute("undeploy googlegsonandavro")
     stmt1.execute("list packages")
     rs = stmt1.getResultSet
-    count = 0
-    if (rs ne null) {
-      while (rs.next()) {
-        count += 1
-      }
-      rs.close()
-    }
-    assert(count == 0)
+    assert(getCount(rs) == 0)
 
     val jarPath = s"$snappyProductDir/jars/hadoop-client-2.7.7.jar"
     stmt1.execute(s"""deploy jar avro-v_1.0 '$jarPath'""")
     stmt1.execute("list jars")
     rs = stmt1.getResultSet
-    count = 0
-    if (rs ne null) {
-      while (rs.next()) {
-        count += 1
-      }
-      rs.close()
-    }
-    assert(count == 1)
+    assert(getCount(rs) == 1)
     stmt1.execute("undeploy  avro-v_1.0 ")
     stmt1.execute("list jars")
     rs = stmt1.getResultSet
-    count = 0
-    if (rs ne null) {
-      while (rs.next()) {
-        count += 1
-      }
-      rs.close()
+    assert(getCount(rs) == 0)
+  }
+  
+  def snap_2772BugTest_deployJar_createExternalTable(): Unit = {
+    logInfo("Running snap_2772BugTest_deployJar_createExternalTable")
+    val user1Conn = getConnection(netPort)
+    val stmt1 = user1Conn.createStatement()
+    stmt1.execute(s"deploy jar cassJar '$cassandraConnectorJarLoc'")
+    stmt1.execute("drop table if exists customer3")
+    stmt1.execute("create external table customer3 using org.apache.spark.sql.cassandra options" +
+        " (table 'customer', keyspace 'test', spark.cassandra.input.fetch.size_in_rows '200000'," +
+        " spark.cassandra.read.timeout_ms '10000')")
+    stmt1.execute("select * from customer3")
+    var rs = stmt1.getResultSet
```

Comment:
Suggestion: `assert(getCount(stmt1.getResultSet) == <value>)` and at other places also.